lib/rules/no-this-alias.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * @fileoverview Disallow aliasing `this`
+ * @author Jed Fox
+ */
+"use strict";
+
+const util = require("../util");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "Disallow aliasing `this`",
+            extraDescription: [util.tslintRule("no-this-assignment")],
+            category: "Best Practices",
+            recommended: false,
+            url: util.metaDocsUrl("no-this-alias"),
+        },
+        fixable: null, // or "code" or "whitespace"
+        schema: [
+            {
+                type: "object",
+                additionalProperties: false,
+                properties: {
+                    allowDestructuring: {
+                        type: "boolean",
+                    },
+                    allowedNames: {
+                        type: "array",
+                        items: {
+                            type: "string",
+                        },
+                    },
+                },
+            },
+        ],
+    },
+
+    create(context) {
+        const { allowDestructuring = false, allowedNames = [] } =
+            context.options[0] || {};
+
+        return {
+            VariableDeclarator(node) {
```

Comment:
```suggestion
            "VariableDeclarator[kind!='type']"(node) {
```

do you mind adding check if it's not type alias?